README.md

```diff
@@ -116,6 +116,24 @@ collaborators:
   # * `maintain` - Recommended for project managers who need to manage the repository without access to sensitive or destructive actions.
   # * `triage` - Recommended for contributors who need to proactively manage issues and pull requests without write access.
 
+# See https://docs.github.com/en/rest/deployments/environments#create-or-update-an-environment for available options
+# Note: deployment_branch_policy differs from the API for ease of use. Either protected_branches (boolean) OR custom_branches (array of strings) is required; this will manage the API requirements under the hood. See https://docs.github.com/en/rest/deployments/branch-policies for documentation of custom_branches. If both are provided in an unexpected manner, protected_branches will be used.
```

Comment:
https://docs.github.com/en/rest/deployments/environments?apiVersion=2022-11-28#create-or-update-an-environment shows an option to allow all branches by setting the `deployment_branch_policy` to `null`. is that enabled with this approach?

Reply:
@rlmartin not sure if you saw this question. i think the rest of the PR looks good, but i think we should handle this if it isnt already.

Reply:
Good question. I'll take a look and get back to you.

Reply:
Good catch - subtle difference between `undefined` and `null` in the GitHub API. I fixed it now and manually verified that it works (and updated the documentation here).